pkg/namesilo_api/resource_record.go

```diff
@@ -0,0 +1,47 @@
+package namesilo_api
+
+import (
+	"encoding/xml"
+)
+
+type ResourceRecord struct {
+	XMLName  xml.Name `xml:"resource_record"`
+	RecordId string   `xml:"record_id"`
+	Type     string   `xml:"type"`
+	Host     string   `xml:"host"`
+	Value    string   `xml:"value"`
+	TTL      int      `xml:"ttl"`
+	Distance int      `xml:"distance"`
+}
+
+func (r ResourceRecord) Equals(other interface{}) bool {
+	if other == nil {
+		return false
+	}
+
+	if r == other {
+		return true
+	}
+
+	var otherRR ResourceRecord
```

Comment:
This is jank AF. Refactor to a second func that accepts a RR only, and ditches the rest of this madness.